### PR TITLE
Show internet protocol on confirmation screens

### PR DESCRIPTION
**Pull request by blackdevelopa**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**
This PR updates the confirmation origin pill. It's a follow up to https://github.com/MetaMask/metamask-mobile/pull/5842

**Screenshots/Recordings**
Before: http://recordit.co/ZahS9jOkLC

After:
1. In-app browser: https://recordit.co/OPpHZeYSns
2. WC: https://recordit.co/Rd8mOlhvvd

_If applicable, add screenshots and/or recordings to visualize the before and after of your change_

**Issue**

Progresses #5842

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by jpuri** [comment]
@blackdevelopa : changes look perfect, can you plz also check if unit test case can be added for this.

**Review by jpuri — APPROVED** [review]

**Review by segun — APPROVED** [review]

**Review by NiranjanaBinoy — APPROVED** [review]

**Comment by seaona** [comment]
When I am trying to deploy a contract MM crashes and I can see `Can't find variable BigInt`. I believe there were [some recent changes](https://consensys.slack.com/archives/C02U025CVU4/p1679445188858539) that cause this behaviour, and may be fixed by resolving the conflicts with main. @blackdevelopa 

https://user-images.githubusercontent.com/54408225/228255247-c1ee2776-ff77-4152-95d7-dc9f6a448238.mp4

**Comment by seaona** [comment]
I can see it fixed. Changes look good @blackdevelopa ! Let's pass again the dev-review and we can merge the PR

https://user-images.githubusercontent.com/54408225/228507802-00180360-bdaf-4656-96d1-f5883cc13785.mp4

File: app/components/UI/ApproveTransactionHeader/ApproveTransactionHeader.test.tsx [hunk 1]
```diff
@@ -1,51 +1,112 @@
...
+      <ApproveTransactionHeader
+        from="0x0"
+        origin="http://metamask.github.io"
+        url="http://metamask.github.io"
+      />,
+      { state: initialState },
+    );
+    expect(getByText('http://metamask.github.io')).toBeDefined();
+  });
 });
```
Comment (jpuri) [inline app/components/UI/ApproveTransactionHeader/ApproveTransactionHeader.test.tsx:112]: 👍

**Review by jpuri — APPROVED** [review]

**Review by NiranjanaBinoy — APPROVED** [review]

**Pull request merged**